lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -883,18 +969,27 @@ def _mean_and_variance_to_probabilities(mean_values, variance_values,
         relative_to_threshold = find_threshold_coordinate(
             probability_cube_template).attributes['spp__relative_to_threshold']
 
-        # Loop over thresholds, and use a normal distribution with the mean
-        # and variance to calculate the probabilities relative to each
+        self._rescale_shape_parameters(
+            mean_values.data.flatten(),
+            np.sqrt(variance_values.data).flatten())
+
+        # Loop over thresholds, and use the specified distribution with the
+        # mean and variance to calculate the probabilities relative to each
         # threshold.
         probabilities = np.empty_like(probability_cube_template.data)
-        distribution = norm(loc=mean_values.data,
-                            scale=np.sqrt(variance_values.data))
+
+        distribution = self.distribution(
+            *self.shape_parameters,
+            loc=mean_values.data.flatten(),
+            scale=np.sqrt(variance_values.data.flatten()))
+
         probability_method = distribution.cdf
         if relative_to_threshold == 'above':
             probability_method = distribution.sf
 
         for index, threshold in enumerate(thresholds):
-            probabilities[index, ...] = probability_method(threshold)
+            probabilities[index, ...] = np.reshape(
+                probability_method(threshold), probabilities.shape[1:])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Value 'probabilities.shape' is unsubscriptable](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4548038)